### PR TITLE
Merge branch '3.x' into master

### PR DESCRIPTION
**Pull request by phansys**

### CODE REVIEW

File: src/Block/Service/AbstractBlockService.php [hunk 1]
```diff
@@ -41,11 +41,14 @@ abstract class AbstractBlockService implements BlockServiceInterface
     public function __construct($twigOrDeprecatedName = null, ?Environment $twig = null)
     {
         if (!$twigOrDeprecatedName instanceof Environment && 0 !== strpos(static::class, __NAMESPACE__.'\\')) {
+            $class = 'c' === static::class[0] && 0 === strpos(static::class, "class@anonymous\0") ? get_parent_class(static::class).'@anonymous' : static::class;
```
Comment (core23) [inline src/Block/Service/AbstractBlockService.php]: This looks very strange oO

Comment (phansys) [reply src/Block/Service/AbstractBlockService.php]: This is already part of 3.x branch:
https://github.com/sonata-project/SonataBlockBundle/blob/38619fa25eda56d32361b6a50e2c5eb6b27b1891/src/Block/Service/AbstractBlockService.php#L61

It's required given the way PHP returns the class name for an anonymous class. See https://github.com/symfony/symfony/pull/29389#discussion_r237944028 for some context.

**Comment by core23** [comment]
There are some conflicts 🤔

**Comment by greg0ire** [comment]
Or at least it says so, but I've seen GH wrongly do this for such manual merge PRs

**Comment by phansys** [comment]
Weird. I see this as "mergeable":
>This branch has no conflicts with the base branch
Merging can be performed automatically.

image

**Comment by core23** [comment]
If you change the merge action to "Rebase and merge", you got the merge error.

**Comment by phansys** [comment]
I must rework the merge, since the commit history was lost after the rebase + cherry-pick.
I'll ping you later.

**Comment by phansys** [comment]
History is back! Sorry for the delay.

**Comment by phansys** [comment]
I can't see the conflict neither, since [the current output](https://travis-ci.org/sonata-project/SonataBlockBundle/jobs/584079894#L381-L396) is not bringing more information.

**Comment by greg0ire** [comment]
I think the problem is that `symfony/cache` is a `psr/cache-implementation`, but `symfony/symfony`  is not (at least in the version we attempt to use). I don't understand why the stable branch build passed though.

**Comment by greg0ire** [comment]
> Rebased! Thank you.

I don't think you should rebase a pull request that is all about a merge. By doing that, you rebase master on stable, which you shouldn't do because you should not rebase a public branch. I think I will help move this forward by creating small PRs that merge stable as it was in the past, a few commits at a time. Should be easier to manage.

**Comment by greg0ire** [comment]
Well actually, I tried `git merge 3.17.0` and it's already too complicated. @phansys , please try `git reset --hard origin/master`, then `git merge 3.17.0`. You have already fixed these conflicts so if you enabled `git rerere` , they should resolve automatically. When this is done, push on your fork, let us review, and then continue with the subsequent commits.

**Comment by phansys** [comment]
Sadly, I didn't have `rerere` enabled. I'll try to start again when I found some time.
Thank you so much for the support.

**Comment by core23** [comment]
Can you please try again @phansys https://github.com/sonata-project/SonataBlockBundle/pull/636 got merged

**Comment by phansys** [comment]
We have more dependency issues [here](https://travis-ci.org/sonata-project/SonataBlockBundle/jobs/589724828#L376-L390).

File: composer.json [hunk 2]
```diff
@@ -22,7 +22,9 @@
     "require": {
         "php": "^7.3",
         "doctrine/common": "^2.3",
+        "psr/cache": "^1.0",
         "sonata-project/cache": "^1.0 || ^2.0",
+        "sonata-project/core-bundle": "^3.15.1",
```
Comment (core23) [inline composer.json]: You should remove this dependency. That’s the reason why we must release new major versions for each bundle

Comment (phansys) [reply composer.json]: My bad :man_facepalming: 
Thank you!

**Comment by core23** [comment]
RTM :100:

**Pull request merged**